File: couchapp/Lib/CouchApplication.py
from com.pittypanda.plugins.couch.glue import CouchApplicationType

import couchapp.dispatch

class CouchApplication(CouchApplicationType, object):

    def __init__(self):
        print 'Initializing'
        pass

    def dispatch(self, args):
        return couchapp.dispatch.dispatch(args)

File: src/com/pittypanda/plugins/couch/glue/CouchApplication.java
package com.pittypanda.plugins.couch.glue;

public class CouchApplication {
  private static CouchApplicationType instance = null;
  
  public static CouchApplicationType getInstance() {
    if (instance == null) {
      JythonObjectFactory factory = new JythonObjectFactory(
          CouchApplicationType.class, "CouchApplication", "CouchApplication"
      );

      instance = (CouchApplicationType) factory.createObject();
    }
    
    return instance;
  }
}
